### ISSUE TITLE
Cognito returns NotAuthorizedException when app client secret is present

### ISSUE THREAD

**Issue by peterallworth** [AlexRudd/cognito-srp#7]
I recently encountered this error when I updated to **v4** from **v2** and supplied a _clientId_ with a non-nil _clientSecret_.
```
operation error Cognito Identity Provider: RespondToAuthChallenge, https response error StatusCode: 400, 
RequestID: e77305a6-8f2a-4a19-9fd6-b420d60d05fa, NotAuthorizedException: Unable to verify secret hash for client...
```
The issue appears to be that the **v2** code uses the username supplied in the challenge parameters for calculating the secret hash whereas the **v4** code uses the username saved in the CognitoSRP structure.

I've tested a fix for this in my fork peterallworth/cognito-srp and it appears to work, as does reverting to **v2**.

**Comment by alexrudd** [comment]
Hi, thanks for spotting this! If you like you can create a PR from your fork and I'll get the fix merged :)

**Comment by peterallworth** [comment]
Hi Alex,

You're welcome. Thank *you* for the package. It's been very useful.
I've completed a PR just now. Let me know if there are any problems.

Regards,
PeterA

**Comment by alexrudd** [comment]
Fixed in #8 and released in v4.1.0

Thanks again!
